gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStubTest.java

```diff
@@ -0,0 +1,168 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.httpjson.longrunning.stub;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.HttpRule;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.httpjson.ApiMethodDescriptor;
+import com.google.api.gax.httpjson.ProtoMessageRequestFormatter;
+import com.google.api.gax.rpc.ClientContext;
+import com.google.api.gax.rpc.testing.FakeCallContext;
+import com.google.common.collect.ImmutableMap;
+import com.google.longrunning.CancelOperationRequest;
+import com.google.longrunning.DeleteOperationRequest;
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.ListOperationsRequest;
+import com.google.protobuf.TypeRegistry;
+import java.io.IOException;
+import java.util.List;
+import org.junit.Test;
+
+public class HttpJsonOperationsStubTest {
+
+  @Test
+  public void testMethodDescriptorsURI() throws IOException {
+    HttpJsonOperationsStub httpJsonOperationsStub =
+        HttpJsonOperationsStub.create(
+            ClientContext.newBuilder()
+                .setCredentials(NoCredentialsProvider.create().getCredentials())
+                .setDefaultCallContext(FakeCallContext.createDefault())
+                .build(),
+            new HttpJsonOperationsCallableFactory(),
+            TypeRegistry.newBuilder().build(),
+            ImmutableMap.of(
+                "google.longrunning.Operations.ListOperations",
+                HttpRule.newBuilder().setGet("testList").build(),
+                "google.longrunning.Operations.GetOperation",
+                HttpRule.newBuilder().setGet("testGet").build(),
+                "google.longrunning.Operations.DeleteOperation",
+                HttpRule.newBuilder().setDelete("testDelete").build(),
+                "google.longrunning.Operations.CancelOperation",
+                HttpRule.newBuilder().setPost("testCancel").build()));
+    // The order is: List, Get, Delete, Cancel
+    List<ApiMethodDescriptor> apiMethodDescriptorList =
+        httpJsonOperationsStub.getAllMethodDescriptors();
+    assertThat(apiMethodDescriptorList.get(0).getRequestFormatter().getPathTemplate().toRawString())
+        .isEqualTo("testList");
```

Comment:
We could extract this string to a variable in the method or static field in the test class, since they are reused in the setup and verification.

Reply:
Done